lib/awful/titlebar.lua

```diff
@@ -453,6 +453,32 @@ local function get_titlebar_function(c, position)
     end
 end
 
+--- Call `request::titlebars` to allow themes or rc.lua to create them even
+-- when `titlebars_enabled` is not set in the rules.
+-- @tparam client c The client.
+-- @tparam[opt=false] boolean hide_all Hide all titlebars except `keep`
+-- @tparam string keep Keep the titlebar at this position
+-- @treturn boolean If the titlebars were loaded
+local function load_titlebars(c, hide_all, keep)
+    if c._request_titlebars_called then return false end
+
+    c:emit_signal("request::titlebars", "awful.titlebar", {})
+
+    if hide_all then
+        -- Don't bother checking if it has been created, `.hide` don't works
+        -- anyway.
+        for _, tb in ipairs {"top", "bottom", "left", "right"} do
+            if tb ~= keep then
+                titlebar.hide(c, tb)
```

Comment:
This is a behaviour change.  If I do `awful.titlebar.show(c, "top")` and then `awful.titlebar.show(c, "bottom")`, the top titlebar is now hidden. Why?!?

Reply:
It wont because this code is only executed once

        if c._request_titlebars_called then return false end

Reply:
Oh, right. Then... why does this hide titlebars that cannot be there?

Reply:
because if toggle or show is called, then the user wants to show one of them. If many are loaded, then they should not be shown.

Reply:
So, this is an intentional change in behaviour then and needs deprecation and has to wait until v5?
Previously, toggling the left titlebar, did not do anything with the other titlebars at all.

Reply:
As far as I see, no, it does not change anything other than fixing the bug. You go from having zero titlebars. Then you want to show one of them. One of them is shown.

Because request::titlebars can potentially load 4 titlebars, the other have to be hidden since they were not requested to be shown.